.circleci/config.yml

```diff
@@ -43,10 +43,10 @@ jobs:
       - when:
           condition:
             not:
-              equal: [ openssl@3, << parameters.OPENSSL_PREINSTALL >> ]
+              equal: [ openssl@3.1, << parameters.OPENSSL_PREINSTALL >> ]
```

Comment:
Why this change? Together with the (unchanged) setting of OPENSSL_PREINSTALL to `openssl@3` in the invocation, this leads to an unnecessary build of OpenSSL(3.2-dev) and a [CI control-flow error below](https://app.circleci.com/pipelines/github/open-quantum-safe/oqs-provider/703/workflows/ac63d142-b2b5-4b60-8e7c-60783b06ec01/jobs/1557). Removing this change should get CI to pass, I'd guess.